modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala

```diff
@@ -0,0 +1,235 @@
+package scala.cli.integration
+
+import com.eed3si9n.expecty.Expecty.expect
+import com.virtuslab.using_directives.config.Settings
+import com.virtuslab.using_directives.custom.model.UsingDirectiveKind
+import com.virtuslab.using_directives.reporter.ConsoleReporter
+import com.virtuslab.using_directives.{Context, UsingDirectivesProcessor}
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.transport.URIish
+
+import scala.jdk.CollectionConverters._
+
+class PublishSetupTests extends munit.FunSuite {
+
+  private def ghUserName = "foo"
+  private def projName   = "project-name"
+  private def devName    = "Alex Test"
+  private def devMail    = "alex@alex.me"
+  private def devUrl     = "https://alex.me"
+
+  private def configSetup(homeDir: os.Path, root: os.Path): Unit = {
+    val dirOptions = Seq[os.Shellable]("--home-directory", homeDir)
+    os.proc(TestUtil.cli, "config", dirOptions, "user.name", devName)
+      .call(cwd = root, stdout = os.Inherit)
+    os.proc(TestUtil.cli, "config", dirOptions, "user.email", devMail)
+      .call(cwd = root, stdout = os.Inherit)
+    os.proc(TestUtil.cli, "config", dirOptions, "user.url", devUrl)
+      .call(cwd = root, stdout = os.Inherit)
+    os.proc(TestUtil.cli, "config", dirOptions, "sonatype.user", "value:uSeR")
+      .call(cwd = root, stdout = os.Inherit)
+    os.proc(TestUtil.cli, "config", dirOptions, "sonatype.password", "value:1234")
+      .call(cwd = root, stdout = os.Inherit)
+    os.proc(TestUtil.cli, "config", dirOptions, "--create-pgp-key")
+      .call(cwd = root, stdout = os.Inherit)
+  }
+
+  private val projDir = os.rel / projName
+  private val testInputs = TestInputs(
+    Seq(
+      os.rel / projDir / "Foo.scala" ->
+        """object Foo
+          |""".stripMargin
```

Comment:
the contents of `Foo.scala` could just be `"object Foo"`, no triple-quote and stripping margins, no?

Reply:
I used that to get an empty line at the end of the file, just in case.

Reply:
ah, okay then.
in that case I'd prefer just:
```scala
"object Foo\n"
```
The alternative is slightly verbose, but I guess it comes down to preference.

Reply:
Yeah, but on Windows that should be `"object Foo\r\n"`, so at the end, we'd get
```scala
"object Foo" + System.lineSeparator()
```
IMO, multiline + strip margin looks slightly better.